wx/lib/checkbox.py

```diff
@@ -0,0 +1,807 @@
+
+        :param `label`: Text to be displayed next to the checkbox.
+        :type `label`: str
+        """
+
+        wx.Control.SetLabel(self, label)
+
+        # The text label has changed, so we must recalculate our best size
+        # and refresh ourselves.
+        self.InvalidateBestSize()
+        self.Refresh()
+
+    def SetFont(self, font):
+        """
+        Sets the :class:`GenCheckBox` text font and updates the control's
+        size to exactly fit the label plus the bitmap.
+
+        :param `font`: Font to be used to render the checkboxs label.
+        :type `font`: `wx.Font`
+        """
+
+        wx.Control.SetFont(self, font)
+
+        # The font for text label has changed, so we must recalculate our best
+        # size and refresh ourselves.
+        self.InvalidateBestSize()
+        self.Refresh()
+
+    def DoGetBestSize(self):
+        """
+        Overridden base class virtual.  Determines the best size of the control
+        based on the label size, the bitmap size and the current font.
+        """
+
+        # Retrieve our properties: the text label, the font and the check
+        # bitmap.
+        label = self.GetLabel()
+        font = self.GetFont()
+        bitmap = self.GetBitmap()
+
+        if not font:
+            # No font defined? So use the default GUI font provided by the system
+            font = self.SYS_DEFAULT_GUI_FONT
+
+        # Set up a wx.ClientDC. When you don't have a dc available (almost
+        # always you don't have it if you are not inside a wx.EVT_PAINT event),
+        # use a wx.ClientDC (or a wx.MemoryDC) to measure text extents.
+        dc = wx.ClientDC(self)
+        dc.SetFont(font)
+
+        # Measure our label.
+        textWidth, textHeight = dc.GetTextExtent(label)
+
+        # Retrieve the check bitmap dimensions.
+        bitmapWidth, bitmapHeight = bitmap.GetWidth(), bitmap.GetHeight()
+
+        # Get the spacing between the check bitmap and the text.
+        spacing = self.GetSpacing()
+
+        # Ok, we're almost done: the total width of the control is simply
+        # the sum of the bitmap width, the spacing and the text width,
+        # while the height is the maximum value between the text width and
+        # the bitmap width.
+        totalWidth = bitmapWidth + spacing + textWidth
+        totalHeight = max(textHeight, bitmapHeight)
+
+        best = wx.Size(totalWidth, totalHeight)
+
+        # Cache the best size so it doesn't need to be calculated again,
+        # at least until some properties of the window change.
+        self.CacheBestSize(best)
+
+        return best
+
+    def AcceptsFocusFromKeyboard(self):
+        """ Overridden base class virtual. """
+
+        # We can accept focus from keyboard, obviously.
+        return True
+
+    def AcceptsFocus(self):
+        """ Overridden base class virtual. """
+
+        # It seems to me that wx.CheckBox does not accept focus with mouse
+        # but please correct me if I am wrong!
```

Comment:
It does. You just can't see the focus rectangle until there's another keypress or navigation event (at least on some platforms.)

Reply:
I changed it to true. False was keeping keyboard focus, but changing to True makes the focus rect always draw. Could be improved if someone wants to be anal about the native appearance 100%. having a focus rect is better than it not working right from spacebar and activating wrong checkbox.